### DISCUSSION TITLE
How to move activity and execute the after automatic?

### DISCUSSION THREAD

**Discussion by lyndonliu89757** [elsa-workflows/elsa-core#4082] [Q&A]
How to move activity in Elsa v3?

Like Flowable RuntimeService:
```
runtimeService.createChangeActivityStateBuilder()
        .processInstanceId("id")
        .moveActivityIdTo("node1", "node2")
        .changeState();
```
```
ChangeActivityStateBuilder moveExecutionToActivityId(String executionId, String activityId);
ChangeActivityStateBuilder moveActivityIdToSubProcessInstanceActivityId(String currentActivityId, String newActivityId, String 
  callActivityId);
```

or Camunda
  https://docs.camunda.org/manual/latest/reference/rest/process-instance/post-modification/
  cancel then startBeforeActivity.

**Comment by sfmskywalker** [comment]
Hello @lyndonliu89757,

Thank you for your detailed question and for providing examples from Flowable and Camunda. As of now, Elsa 3 does not have built-in support for moving the execution pointer. This functionality would likely be implemented within the Flowchart activity, which orchestrates its child activities. However, I want to assure you that we are aware of this requirement and have it on our radar for future enhancements.

We appreciate your feedback and patience as we continue to improve Elsa. If you have any other questions or suggestions, please don't hesitate to share.

**Reply by sfmskywalker** [reply]
I created [this issue](https://github.com/elsa-workflows/elsa-core/issues/4427) to track this requirement.
